pykusto/client.py

```diff
@@ -30,16 +33,23 @@ def __init__(self, client_or_cluster: Union[str, KustoClient]) -> None:
         else:
             self._client = self._get_client_for_cluster(client_or_cluster)
             self._cluster_name = client_or_cluster
+        self._items = None
+        super().__init__(retrieve_by_default)
```

Comment:
Calling the super's c'tor not in the beginning indicate there's something fishy in the initialisation process.

Reply:
Indeed. I re-wrote the code and was able to avoid this in `Database` and `Table`, but in `PyKustoClient` there is still one line needed before the call to super in order to initialize the client. I think this is reasonable and couldn't think of a better alternative. Let me know if you can think of something.